packages/apm/src/integrations/tracing.ts

```diff
@@ -336,11 +343,171 @@ export class Tracing implements Integration {
+      Tracing._addPerformanceEntries(active);
       // true = use timestamp of last span
       active.finish(true);
     }
   }
 
+  /**
+   * This uses `performance.getEntries()` to add additional spans to the active transaction.
+   * Also, we update our timings since we consider the timings in this API to be more correct than our manual
+   * measurements.
+   *
+   * @param transactionSpan The transaction span
+   */
+  private static _addPerformanceEntries(transactionSpan: SpanClass): void {
+    let navigationOffset = 0;
+    if (
+      transactionSpan.op === 'navigation' &&
+      transactionSpan.data &&
+      typeof transactionSpan.data.offset === 'number'
+    ) {
+      navigationOffset = transactionSpan.data.offset;
+    }
+    // tslint:disable-next-line: completed-docs
+    function addSpan(span: SpanClass): void {
+      // tslint:disable-next-line: no-non-null-assertion
+      transactionSpan.spanRecorder!.finishSpan(span);
+    }
+
+    // tslint:disable-next-line: completed-docs
+    function addPerformanceNavigationTiming(parent: SpanClass, entry: any, event: string): void {
+      const span = parent.child({
+        description: event,
+        op: 'browser',
+      });
+      // tslint:disable: no-unsafe-any
+      span.startTimestamp = parent.startTimestamp + Tracing._msToSec(entry[`${event}Start`] as number);
+      span.timestamp = parent.startTimestamp + Tracing._msToSec(entry[`${event}End`] as number);
+      // tslint:enable: no-unsafe-any
+      addSpan(span);
+    }
+
+    // tslint:disable-next-line: completed-docs
+    function addRequest(parent: SpanClass, entry: any): void {
+      const request = parent.child({
+        description: 'request',
+        op: 'browser',
+      });
+      // tslint:disable: no-unsafe-any
+      request.startTimestamp = parent.startTimestamp + Tracing._msToSec(entry.requestStart as number);
+      request.timestamp = parent.startTimestamp + Tracing._msToSec(entry.responseEnd as number);
+      // tslint:enable: no-unsafe-any
+      addSpan(request);
+      const response = parent.child({
+        description: 'response',
+        op: 'browser',
+      });
+      // tslint:disable: no-unsafe-any
+      response.startTimestamp = parent.startTimestamp + Tracing._msToSec(entry.responseStart as number);
+      response.timestamp = parent.startTimestamp + Tracing._msToSec(entry.responseEnd as number);
+      // tslint:enable: no-unsafe-any
+      addSpan(response);
+    }
+
+    let entryScriptSrc: string | undefined;
+
+    if (global.document) {
+      // tslint:disable-next-line: prefer-for-of
+      for (let i = 0; i < document.scripts.length; i++) {
+        // We go through all scripts on the page and look for 'data-entry'
+        // We remember the name and measure the time between this script finished loading and
+        // our mark 'sentry-tracing-init'
+        if (document.scripts[i].getAttribute('data-entry') === 'true') {
```

Comment:
Does the user need to manually indicate the `data-entry` attribute?

Reply:
Yes, otherwise it's hard to determine which is the "entry script" of your application.
If that `data-entry` is not there, it's just missing the `script.evaluation` span.